src/Passes/Source/Rules/Notation/Notation.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Rules/Notation/Notation.hpp"
+#include "Rules/Operands/Any.hpp"
+#include "Rules/Operands/Call.hpp"
+#include "Rules/Operands/Instruction.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+    namespace notation
+    {
+
+        std::function<bool(
+            ReplacementRule::Builder&,
+            ReplacementRule::Value*,
+            ReplacementRule::Captures&,
+            ReplacementRule::Replacements&)>
+        deleteInstruction()
+        {
+            return [](ReplacementRule::Builder&, ReplacementRule::Value* val, ReplacementRule::Captures&,
```

Comment:
For some of these rules it might be helpful to have a comment with an example of the pattern it is meant to recognize. This on in particular is a little trickier to understand at first glance.

Reply:
I've added a comment explaining the purpose of this function: It is not a rule, but a shorthand notation the produces the logic to delete and instruction. That is to say, it is a replacer function factory that does not contain any patterns. An example use is following: ```c++
addRule({callByNameOnly(name), deleteInstruction()});
```